src/shared/azureagent/resources.go

```diff
@@ -0,0 +1,70 @@
+package azureagent
+
+import (
+	"context"
+	"fmt"
+	"github.com/otterize/intents-operator/src/shared/errors"
+	"regexp"
+	"strings"
+)
+
+const StorageAccountPattern = `providers/Microsoft.Storage/storageAccounts/([^/]+)`
+
+func (a *Agent) ValidateScope(ctx context.Context, scope string) error {
+	res, err := a.resourceClient.GetByID(ctx, scope, "2022-09-01", nil)
+	if err != nil {
+		return err
+	}
+	if res.GenericResource.ID == nil {
+		return errors.Errorf("scope %s not found", scope)
+	}
+	return nil
+}
+
+func (a *Agent) GetFullResourceScope(ctx context.Context, partialScope string) (string, error) {
+	subscriptions, err := a.ListSubscriptions(ctx)
+	if err != nil {
+		return "", errors.Wrap(err)
+	}
+
+	// Storage accounts are unique across azure, so we should look up the storage account directly
+	regex := regexp.MustCompile(StorageAccountPattern)
+	matchStorageAccount := regex.FindStringSubmatch(partialScope)
+	scopeToCheck := partialScope
+	if len(matchStorageAccount) > 1 {
+		scopeToCheck = fmt.Sprintf("/providers/Microsoft.Storage/storageAccounts/%s", matchStorageAccount[1])
+	}
+
+	for _, sub := range subscriptions {
```

Comment:
This may take a really long time in a world where there's many subscriptions with many resources in them. 
Some ideas: 
1. use `resourceClient.NewListPager` with filter, to filter by resource type (storage accounts)
2. add some local cache and store resolved storage accounts to avoid searching for them over and over again